commons-executors/src/main/java/com/palantir/common/concurrent/PTExecutors.java

```diff
@@ -501,7 +487,13 @@ public void run() {
                     ExecutorInheritableThreadLocal.uninstallMapOnThread(oldMap);
                 }
             }
-        };
+        });
+        if (runnable instanceof Future<?>) {
+            @SuppressWarnings("unchecked")
+            Future<Object> unsafeFuture = (Future<Object>) runnable;
+            return new ForwardingRunnableFuture<Object>(wrappedRunnable, unsafeFuture);
+        }
+        return wrappedRunnable;
     }
 
     public static <T> RunnableFuture<T> wrap(RunnableFuture<T> rf) {
```

Comment:
I don't think we use this method, but let's also wrap it just to be safe.

Reply:
Thanks for catching that - fixed it and refactored the code a bit more.